src/components/Chart/AmpereChart.tsx

```diff
@@ -66,25 +117,41 @@ const AmpereChart = ({
     } = useSelector(chartState);
     const { samplingRunning } = useSelector(appState);
     const isRealTimePane = useSelector(isRealTimePaneSelector);
-    const sendTriggerLevel = level => dispatch(updateTriggerLevelAction(level));
-    const updateTriggerLevel = level =>
+    const sendTriggerLevel = (level: number) =>
+        dispatch(updateTriggerLevelAction(level));
+    const updateTriggerLevel = (level: number) =>
         dispatch(triggerLevelSetAction({ triggerLevel: level }));
 
+    const formatCurrent = (uA: number) =>
+        typeof uA === 'number'
+            ? unit(uA, 'uA')
+                  .format({ notation: 'auto', precision: 4 })
+                  .replace('u', '\u00B5')
+            : (undefined as never);
+
     const timestampToLabel = React.useCallback(
         (_usecs, index, array) => {
+            if (typeof _usecs !== 'number') {
+                return undefined as never;
+            }
             let usecs = _usecs;
-            if (triggerOrigin != null) {
-                usecs -= indexToTimestamp(triggerOrigin);
+            const timestampAtTriggerOrigin =
+                triggerOrigin == null ? null : indexToTimestamp(triggerOrigin);
+            if (timestampAtTriggerOrigin != null) {
+                usecs -= timestampAtTriggerOrigin;
             }
             const microseconds = Math.abs(usecs);
             const sign = usecs < 0 ? '-' : '';
             if (!array) {
                 return `${sign}${Number(microseconds / 1e3).toFixed(3)} ms`;
             }
-            if (index > 0 && index < array.length - 1) {
-                const first = array[0] - indexToTimestamp(triggerOrigin);
-                const last =
-                    array[array.length - 1] - indexToTimestamp(triggerOrigin);
+            if (
+                timestampAtTriggerOrigin &&
+                index > 0 &&
+                index < array.length - 1
+            ) {
+                const first = array[0] - timestampAtTriggerOrigin;
```

Comment:
What's in this `array`? I know it was already named this before this pr, but could it be named something like `timeStamps` or similar?